hazelcast/src/main/java/com/hazelcast/internal/cluster/impl/ClusterServiceImpl.java

```diff
@@ -233,7 +233,7 @@ void doRemoveAddress(Address deadAddress, boolean destroyConnection) {
             }
             Connection conn = node.connectionManager.getConnection(deadAddress);
             if (destroyConnection && conn != null) {
-                node.connectionManager.destroyConnection(conn);
+                conn.close(reason, null);
```

Comment:
The old design was problematic due to have 2 entrances for closing a connection. now it has been simplified; everything goes through the connection.close; which does a callback on the connectionmanager to do whatever else needs to be done. Also removes potential races since the connection.close provides a guarantee that the connection will be closed once.